File: Calculadora.java

import java.util.Scanner;

public class Calculadora {
    public static double sumar(double a, double b) {
            return a + b;
        }
    public static double restar(double a, double b) {
            return a - b;
        }
    public static double multiplicacion(double a, double b) {
            return a * b;
        }
    public static double  division(double a, double b) {
            return a / b;
        }

    public static void main(String[] args) {

        Scanner scanner = new Scanner(System.in);
        
        int i = -1;
        while (i != 0) {
            System.out.println("1. Suma");
            System.out.println("2. Restar");
            System.out.println("3. Multiplicar");
            System.out.println("4. Dividir");
            System.out.println("0. Salir");
            i = scanner.nextInt();

            //Suma
            if (i == 1) {
                System.out.println("Primer digito");
                double a = scanner.nextDouble();
                System.out.println("Segundo digito");
                double b = scanner.nextDouble();
                System.out.println(sumar(a, b));
            }

            //Resta
            else if (i == 2) {
                System.out.println("Primer digito");
                double a = scanner.nextDouble();
                System.out.println("Segundo digito");
                double b = scanner.nextDouble();
                System.out.println(restar(a, b));
            }

            //Multiplicacion
            else if (i == 3) {
                System.out.println("Primer digito");
                double a = scanner.nextDouble();
                System.out.println("Segundo digito");
                double b = scanner.nextDouble();
                System.out.println(multiplicacion(a, b));
            }

            //Division
            else if (i == 4) {
                System.out.println("Primer digito");
                double a = scanner.nextDouble();
                System.out.println("Segundo digito");
                double b = scanner.nextDouble();

                if(b==0) {
                    System.out.println("Error: no se puede dividir entre 0");
                }else {
                    System.out.println(division(a, b));
                }
            }

            // Si el usuario inserta otro entero que no es de la lista
            else {
                System.out.println("Por favor introducir un digito valido de la lista.");
            }
        }
        // Cierra el scanner
        scanner.close();
    }
    
}

File: Calculadora.py
def sumar(a, b):
    return a + b


def restar(a, b):
    return a - b


def multiplicacion(a, b):
    return a * b


def division(a, b):
    return a / b


i = -1
while i != 0:
    print("1. Suma\n" "2. Resta\n" "3. Multiplicacion\n" "4. Division\n" "0. Salir")
    i = int(input())

    if i == 1:
        a = float(input())
        b = float(input())
        print(sumar(a, b))
    elif i == 2:
        a = float(input())
        b = float(input())
        print(restar(a, b))
    elif i == 3:
        a = float(input())
        b = float(input())
        print(multiplicacion(a, b))
    elif i == 4:
        a = float(input())
        b = float(input())

        if b == 0:
            print("No se puede dividir entre 0.")
        else:
            print(division(a, b))

    else:
        print("Escoge un entero de la lista.")

print("Gracias por usar el programa")
